Index: app/src/main/java/org/dci/walletapp/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package org.dci.walletapp;\n\nimport android.os.Bundle;\n\nimport androidx.activity.EdgeToEdge;\nimport androidx.appcompat.app.AppCompatActivity;\nimport androidx.core.graphics.Insets;\nimport androidx.core.view.ViewCompat;\nimport androidx.core.view.WindowInsetsCompat;\n\npublic class MainActivity extends AppCompatActivity {\n\n    @Override\n    protected void onCreate(Bundle savedInstanceState) {\n        super.onCreate(savedInstanceState);\n        EdgeToEdge.enable(this);\n        setContentView(R.layout.activity_main);\n        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {\n            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());\n            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);\n            return insets;\n        });\n    }\n}
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/org/dci/walletapp/MainActivity.java b/app/src/main/java/org/dci/walletapp/MainActivity.java
--- a/app/src/main/java/org/dci/walletapp/MainActivity.java	(revision 725dea1ffbc863cd04842c13f79d29326d97567e)
+++ b/app/src/main/java/org/dci/walletapp/MainActivity.java	(date 1719840611644)
@@ -20,5 +20,15 @@
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
             return insets;
         });
+        test();
+    }
+
+
+
+
+    private void test() {
+        Transaction transaction = new Transaction(2.0, "Description", true, "whatever");
+        JsonFilesOperations filesOperations = new JsonFilesOperations();
+        filesOperations.write(this, transaction);
     }
 }
\ No newline at end of file
Index: app/src/main/java/org/dci/walletapp/JsonFilesOperations.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/org/dci/walletapp/JsonFilesOperations.java b/app/src/main/java/org/dci/walletapp/JsonFilesOperations.java
new file mode 100644
--- /dev/null	(date 1719840685236)
+++ b/app/src/main/java/org/dci/walletapp/JsonFilesOperations.java	(date 1719840685236)
@@ -0,0 +1,46 @@
+package org.dci.walletapp;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class JsonFilesOperations {
+
+
+    public void write(Context context, Transaction transaction) {
+        String userString = convertTransaction(transaction).toString();
+
+        File file = new File(context.getFilesDir(), "transactions.json");
+
+        try (FileWriter fileWriter = new FileWriter(file);
+             BufferedWriter bufferedWriter = new BufferedWriter(fileWriter)) {
+            bufferedWriter.write(userString);
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        Log.d("tetjson", file.getAbsolutePath());
+    }
+
+    private JSONObject convertTransaction(Transaction transaction) {
+        JSONObject jsonObject = new JSONObject();
+        try {
+            jsonObject.put("isIncome", String.valueOf(transaction.isIncome()));
+            jsonObject.put("amount", transaction.getAmount());
+            jsonObject.put("dateTime", transaction.getDateTime());
+            jsonObject.put("incomeSource", transaction.getSource());
+            jsonObject.put("description", transaction.getDescription());
+        } catch (JSONException e) {
+            throw new RuntimeException(e);
+        }
+
+        return jsonObject;
+    }
+}
\ No newline at end of file
Index: app/src/main/java/org/dci/walletapp/Transaction.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/org/dci/walletapp/Transaction.java b/app/src/main/java/org/dci/walletapp/Transaction.java
new file mode 100644
--- /dev/null	(date 1719840617316)
+++ b/app/src/main/java/org/dci/walletapp/Transaction.java	(date 1719840617316)
@@ -0,0 +1,71 @@
+package org.dci.walletapp;
+
+import android.os.Build;
+
+import java.time.LocalDateTime;
+
+public class Transaction {
+    private boolean income;
+    private double amount;
+    private LocalDateTime dateTime;
+    private String source;
+    private String description;
+
+    public Transaction(double amount, String description, boolean income, String source) {
+        this.amount = amount;
+        this.description = description;
+        this.income = income;
+        this.source = source;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            dateTime = LocalDateTime.now();
+        }
+    }
+
+    public Transaction(double amount, LocalDateTime dateTime, String description, boolean income, String source) {
+        this.amount = amount;
+        this.dateTime = dateTime;
+        this.description = description;
+        this.income = income;
+        this.source = source;
+    }
+
+    public double getAmount() {
+        return amount;
+    }
+
+    public void setAmount(double amount) {
+        this.amount = amount;
+    }
+
+    public LocalDateTime getDateTime() {
+        return dateTime;
+    }
+
+    public void setDateTime(LocalDateTime dateTime) {
+        this.dateTime = dateTime;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public boolean isIncome() {
+        return income;
+    }
+
+    public void setIncome(boolean income) {
+        this.income = income;
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public void setSource(String incomeSource) {
+        this.source = incomeSource;
+    }
+}
\ No newline at end of file
